### PR TITLE
feat: Add TS 5.0 options

### PR DESCRIPTION
**Pull request by shun-shobon**
Hi. I have added an option to tsconfig.json that was added by the recently released TypeScript 5.0.

Ref: https://devblogs.microsoft.com/typescript/announcing-typescript-5-0

### CODE REVIEW

File: source/tsconfig-json.d.ts [hunk 1]
```diff
@@ -956,6 +962,51 @@ declare namespace TsConfigJson {
...
+		/**
+		Anything that uses the type modifier is dropped entirely.
+
+		@default false
+		*/
+		verbatimModuleSyntax?: boolean;
+
+		/**
+		Suppress deprecation warnings
+		 */
```
Comment (sindresorhus) [inline source/tsconfig-json.d.ts:1008]:
```suggestion
		*/
```

**Comment by sindresorhus** [comment]
Thanks for working on this. Can you double check that you covered all the changes for 5.0?

**Comment by shun-shobon** [comment]
@sindresorhus 
I checked it. Can you review this?

**Comment by kidonng** [comment]
According to https://github.com/SchemaStore/schemastore/pull/2754/files#diff-c2958f962599020051d4dcbd1983284db50f59857bf58b26b24da6d6a763a257R572, the `Module` type should add `ES2023`.

**Comment by shun-shobon** [comment]
@kidonng 

I have tested and it appears that TS5.0.2 does not support ES2023.

image

**Comment by kidonng** [comment]
That's weird, because there are definitions for ES2023: https://github.com/microsoft/TypeScript/blob/v5.0.2/lib/lib.es2023.d.ts
Some TS internals might be outdated I guess.

**Comment by shun-shobon** [comment]
When I was actually running TypeScript, it seems that the `moduleResolution` option `Node10` was added to the `moduleResolution`. There's a lot I don't understand since the documentation hasn't been updated yet...

https://github.com/microsoft/TypeScript/pull/51901

**Comment by michalbundyra** [comment]
@sindresorhus As it is merged, can we expect a release soon? Or something else needs to be done to provide support for TypeScript 5.0? Thanks

**Pull request merged**